integration_test/testdata/otlp/metrics.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+func installMetricExportPipeline(ctx context.Context) (func(context.Context) error, error) {
+	exporter, err := otlpmetricgrpc.New(ctx, otlpmetricgrpc.WithInsecure())
+	if err != nil {
+		log.Fatal(err)
+	}
+	metricProvider := metric.NewMeterProvider(
+		metric.WithReader(metric.NewPeriodicReader(exporter)),
+		metric.WithResource(resource.Default()),
+	)
+	global.SetMeterProvider(metricProvider)
+	return metricProvider.Shutdown, nil
+}
+
+func main() {
+	ctx := context.Background()
+
+	shutdown, err := installMetricExportPipeline(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer func() {
+		if err := shutdown(ctx); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	meter := global.MeterProvider().Meter("foo")
+	gauge, err := meter.AsyncFloat64().Gauge("otlp.test")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = meter.RegisterCallback([]instrument.Asynchronous{gauge}, func(c context.Context) {
```

Comment:
Is it enough to test with a `GAUGE` metric ? Maybe the flake was due to the timestamps in a `CUMULUATIVE` are different to a `GAUGE` (start and end are the same).

Reply:
The flakes were really weird; for some reason it only happened on Windows, and only sometimes, and I could make it pass more often by adding a simple delay at the end before tearing it down. When I dug into the backend I noticed that GCM was trying to connect two time series points together which were sent on totally separate runs. So my guess is that it needs to either send two points in quick succession to work, or the program needs to live long enough to create its own delta, or something like that.

I figure GAUGE is more reliable at any rate since it doesn't require any delta shenanigans. It is a good point though that we could be testing all supported metric types, but I'd have to root cause the flake issue first.

Reply:
Looks like the CUMULATIVE issue was caused by only sending one point -- if I send a second point with a short delay in between it seems to work reliably.

There also doesn't seem to be a way to send DELTA metrics so I'll leave it at GAUGE and CUMULATIVE for now.

Reply:
Awesome!